### PR TITLE
dm: update DM config to integrate with lightning

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

update DM config to integrate with lightning.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/12296
- Other reference link(s):

### CODE REVIEW

File: dm/dm-precheck.md [hunk 1]
```diff
@@ -90,6 +90,30 @@ For the full data migration mode (`task-mode: full`), in addition to the [common
...
+
+* Empty Regions in the downstream database
+
+    - If the number of empty Regions is greater than `max(1000, 3 * the number of tables)` (the larger of "1000" and "3 times the number of tables"), the precheck returns a warning. You can adjust related PD parameters to speed up the merging of empty Regions and wait for the number of empty Regions to decrease. See [PD Scheduling Best Practices - Slow Region Merge](/best-practices/pd-scheduling-best-practices.md#region-merge-is-slow).
+
+* Region distribution in the downstream database
+
+    - Checks the number of Regions in different TiKV nodes. If the node with the least number of Regions has a number of Regions that is 75% of the node with the most number of Regions, the precheck returns a warning. You can adjust related PD parameters to speed up the scheduling of Regions and wait for the number of Regions to change. See [PD Scheduling Best Practices - Leader/Region distribution is not balanced](/best-practices/pd-scheduling-best-practices.md#leadersregions-are-not-evenly-distributed).
```
Comment (lance6716) [inline dm/dm-precheck.md:103]: > If the node with the least number of Regions has a number of Regions that is 75% of the node with the most number of Regions

hard to read 😂 can we change the subject to number (of regions on node) and refine it?

File: dm/dm-precheck.md [hunk 2]
```diff
@@ -90,6 +90,30 @@ For the full data migration mode (`task-mode: full`), in addition to the [common
...
+
+    - Physical import must call the interfaces of TiDB, PD, and TiKV. If the version is not compatible, the precheck returns an error.
+
+* The free space of the downstream database
+
+    - Estimates the sum of the sizes of all tables in the allow list in the upstream database (`source_size`). If the free space of the downstream database is less than `source_size`, the precheck returns an error. If the free space of the downstream database is less than the number of TiKV replicas \* `source_size` \* 2, the precheck returns a warning.
+
+* Whether the downstream database is running tasks that are incompatible with physical import
+
+    - Currently, physical import is incompatible with [TiCDC](/ticdc/ticdc-overview.md) and [PITR](/br/br-pitr-guide.md) tasks. If these tasks are running in the downstream database, the precheck returns an error.
```
Comment (lance6716) [inline dm/dm-precheck.md:115]:
```suggestion
    - Currently, physical import is incompatible with [TiCDC](/ticdc/ticdc-overview.md) and [PiTR](/br/br-pitr-guide.md) tasks. If these tasks are running in the downstream database, the precheck returns an error.
```

maybe it's "Point in TIme Recovery"

File: dm/task-configuration-file-full.md [hunk 3]
```diff
@@ -120,15 +120,45 @@ loaders:
...
+    #   Conflicting data is not resolved in this method. "none" has the best performance, but
+    #   might lead to inconsistent data in the downstream database.
+    # - "manual". Corresponds to the "remove" strategy of conflict detection in TiDB Lightning's physical import.
+    #   (https://docs.pingcap.com/tidb/stable/tidb-lightning-physical-import-mode-usage#conflict-detection).
+    #   When the import encounter conflicting data, DM removes all conflicting records from
+    #   the target table and records the data in the `${meta-schema}_${name}.conflict_error_v1`
+    #   table. In this configuration file, the conflicting data is recorded in the
+    #   `dm_meta_test.conflict_error_v1` table. When the full import phase is completed, the
+    #   tasks is paused and you are prompted to query this table and manually resolve the
+    #   conflicts. You need to resume the task and enter the incremental phase using the #   `resume-task` command.
```
Comment (lance6716) [inline dm/task-configuration-file-full.md:146]:
```suggestion
    #   conflicts. You need to resume the task and enter the incremental phase using the `resume-task` command.
```

File: dm/dm-precheck.md (hunk 2 above)
Comment (ran-huang) [reply dm/dm-precheck.md:115]: Actually, this has been discussed a lot. 😂 
We used "PiTR" in the first place but later switched to "PITR", as it is consistent with [Wikipedia](https://en.wikipedia.org/wiki/Point-in-time_recovery) and [AWS documentation](https://aws.amazon.com/cn/dynamodb/pitr/).

File: dm/dm-precheck.md (hunk 1 above)
Comment (ran-huang) [inline dm/dm-precheck.md:103]: Thanks... It is indeed bad English. How about this? 

```suggestion
    - Checks the number of Regions on different TiKV nodes. If the number of Regions on one node is less than 75% of the number of Regions on any other node, the precheck returns a warning. You can adjust related PD parameters to speed up the scheduling of Regions and wait for the number of Regions to change. See [PD Scheduling Best Practices - Leader/Region distribution is not balanced](/best-practices/pd-scheduling-best-practices.md#leadersregions-are-not-evenly-distributed).
```

Comment (TomShawn) [reply dm/dm-precheck.md:103]: Aggree. It's a bit hard to read. Can we state the meaning in this way? 
 (I am not sure whether it's correct) https://github.com/pingcap/docs-cn/pull/12296/files#r1095384413
image

File: dm/dm-precheck.md [hunk 4]
```diff
@@ -90,6 +90,30 @@ For the full data migration mode (`task-mode: full`), in addition to the [common
...
+
+* Empty Regions in the downstream database
+
+    - If the number of empty Regions is greater than `max(1000, 3 * the number of tables)` (the larger of "1000" and "3 times the number of tables"), the precheck returns a warning. You can adjust related PD parameters to speed up the merging of empty Regions and wait for the number of empty Regions to decrease. See [PD Scheduling Best Practices - Slow Region Merge](/best-practices/pd-scheduling-best-practices.md#region-merge-is-slow).
+
+* Region distribution in the downstream database
+
+    - Checks the number of Regions in different TiKV nodes. If the node with the least number of Regions has a number of Regions that is 75% of the node with the most number of Regions, the precheck returns a warning. You can adjust related PD parameters to speed up the scheduling of Regions and wait for the number of Regions to change. See [PD Scheduling Best Practices - Leader/Region distribution is not balanced](/best-practices/pd-scheduling-best-practices.md#leadersregions-are-not-evenly-distributed).
+
+* The version of TiDB, PD, and TiKV in the downstream database
```
Comment (TomShawn) [inline dm/dm-precheck.md:105]:
```suggestion
* The versions of TiDB, PD, and TiKV in the downstream database
```

File: dm/dm-precheck.md [hunk 5]
```diff
@@ -90,6 +90,30 @@ For the full data migration mode (`task-mode: full`), in addition to the [common
...
+
+    - Physical import must call the interfaces of TiDB, PD, and TiKV. If the version is not compatible, the precheck returns an error.
```
Comment (TomShawn) [inline dm/dm-precheck.md:107]:
```suggestion
    - Physical import must call the interfaces of TiDB, PD, and TiKV. If the versions are not compatible, the precheck returns an error.
```

File: dm/dm-precheck.md [hunk 6]
```diff
@@ -130,6 +154,11 @@ Prechecks can find potential risks in your environments. It is not recommended t
 | `table_schema`          | Checks the compatibility of the table schemas in the upstream MySQL tables. |
 | `schema_of_shard_tables`| Checks the consistency of the table schemas in the upstream MySQL multi-instance shards. |
 | `auto_increment_ID`     | Checks whether the auto-increment primary key conflicts in the upstream MySQL multi-instance shards. |
+| `empty_region` | Checks the number of empty Regions in the downstream database for physical import. |
+| `region_distribution` | Checks the distribution of Regions in the downstream database for physical import. |
+| `downstream_version` | Checks the version of TiDB, PD, and TiKV in the downstream database. |
```
Comment (TomShawn) [inline dm/dm-precheck.md:159]:
```suggestion
| `downstream_version` | Checks the versions of TiDB, PD, and TiKV in the downstream database. |
```

File: dm/monitor-a-dm-cluster.md [hunk 7]
```diff
@@ -62,15 +62,14 @@ The following metrics show only when `task-mode` is in the `full` or `all` mode.
 
 | Metric name | Description | Alert | Severity level |
 |:----|:------------|:----|:----|
+| dump progress | The percentage of the completed dumping process of the dump unit. The value range is 0%~100% | N/A | N/A |
 | load progress | The percentage of the completed loading process of the load unit. The value range is 0%~100% | N/A | N/A |
+| checksum progress | The percentage of the completed checksum process after the load unit finishes dumping. The value range is 0%~100% | N/A | N/A |
+| total bytes for load unit | The bytes processed in the parsing, generating data KV, and generating index KV stages of the import process by the load unit | N/A | N/A |
```
Comment (TomShawn) [inline dm/monitor-a-dm-cluster.md:68]:
```suggestion
| total bytes for load unit | The bytes processed in the parsing, in generating data KV, and in generating index KV stages of the import process by the load unit | N/A | N/A |
```

File: dm/task-configuration-file-full.md [hunk 8]
```diff
@@ -120,15 +120,45 @@ loaders:
...
-    # - "loader". Use Loader mode to import data. This mode is only for compatibility with features that TiDB Lightning does not support yet. It will be deprecated in the future.
-    import-mode: "sql"
-    #  Methods to resolve conflicts during the full import phase. You can set it to the following:
-    # - "replace" (default). Only supports the import mode "sql". In this method, it uses the new data to replace the existing data.
-    # - "ignore". Only supports the import mode "sql". It keeps the existing data, and ignores the new data.
-    # - "error". Only supports the import mode "loader". It reports errors when inserting duplicated data, and then stops the replication task.
-    on-duplicate: "replace"
+
+    # The import mode during the full import phase. The following modes are supported:
+    # - "logical" (default). Use TiDB Lightning's logical import mode to import data. Document: https://docs.pingcap.com/tidb/stable/tidb-lightning-logical-import-mode
```
Comment (TomShawn) [inline dm/task-configuration-file-full.md:125]:
```suggestion
    # - "logical" (default). Uses TiDB Lightning's logical import mode to import data. Document: https://docs.pingcap.com/tidb/stable/tidb-lightning-logical-import-mode
```

File: dm/task-configuration-file-full.md [hunk 9]
```diff
@@ -120,15 +120,45 @@ loaders:
...
+    # - "physical". Use TiDB Lightning's physical import mode to import data. Document: https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-physical-import-mode
```
Comment (TomShawn) [inline dm/task-configuration-file-full.md:126]:
```suggestion
    # - "physical". Uses TiDB Lightning's physical import mode to import data. Document: https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-physical-import-mode
```

File: dm/task-configuration-file-full.md [hunk 10]
```diff
@@ -120,15 +120,45 @@ loaders:
...
+    # - "manual". Corresponds to the "remove" strategy of conflict detection in TiDB Lightning's physical import.
+    #   (https://docs.pingcap.com/tidb/stable/tidb-lightning-physical-import-mode-usage#conflict-detection).
+    #   When the import encounter conflicting data, DM removes all conflicting records from
+    #   the target table and records the data in the `${meta-schema}_${name}.conflict_error_v1`
+    #   table. In this configuration file, the conflicting data is recorded in the
+    #   `dm_meta_test.conflict_error_v1` table. When the full import phase is completed, the
+    #   tasks is paused and you are prompted to query this table and manually resolve the
+    #   conflicts. You need to resume the task and enter the incremental phase using the `resume-task` command.
+    on-duplicate-physical: "none"
+    # The directory of local KV sorting in the physical import mode. The default value of this
```
Comment (TomShawn) [inline dm/task-configuration-file-full.md:148]:
```suggestion
    # The directory used for local KV sorting in the physical import mode. The default value of this
```

File: dm/dm-precheck.md [hunk 11]
```diff
@@ -90,6 +90,30 @@ For the full data migration mode (`task-mode: full`), in addition to the [common
...
+
+    - Checks the number of Regions in different TiKV nodes. If the node with the least number of Regions has a number of Regions that is 75% of the node with the most number of Regions, the precheck returns a warning. You can adjust related PD parameters to speed up the scheduling of Regions and wait for the number of Regions to change. See [PD Scheduling Best Practices - Leader/Region distribution is not balanced](/best-practices/pd-scheduling-best-practices.md#leadersregions-are-not-evenly-distributed).
+
+* The version of TiDB, PD, and TiKV in the downstream database
+
+    - Physical import must call the interfaces of TiDB, PD, and TiKV. If the version is not compatible, the precheck returns an error.
+
+* The free space of the downstream database
+
+    - Estimates the sum of the sizes of all tables in the allow list in the upstream database (`source_size`). If the free space of the downstream database is less than `source_size`, the precheck returns an error. If the free space of the downstream database is less than the number of TiKV replicas \* `source_size` \* 2, the precheck returns a warning.
```
Comment (TomShawn) [inline dm/dm-precheck.md:111]:
```suggestion
    - Estimates the total sizes of all tables in the allow list in the upstream database (`source_size`). If the free space of the downstream database is less than `source_size`, the precheck returns an error. If the free space of the downstream database is less than the number of TiKV replicas \* `source_size` \* 2, the precheck returns a warning.
```

**Review by TomShawn — COMMENTED** [review]
mostly minor comments

**Review by TomShawn — APPROVED** [review]
LGTM

File: dm/task-configuration-file-full.md [hunk 12]
```diff
@@ -120,15 +120,45 @@ loaders:
...
-    import-mode: "sql"
-    #  Methods to resolve conflicts during the full import phase. You can set it to the following:
-    # - "replace" (default). Only supports the import mode "sql". In this method, it uses the new data to replace the existing data.
-    # - "ignore". Only supports the import mode "sql". It keeps the existing data, and ignores the new data.
-    # - "error". Only supports the import mode "loader". It reports errors when inserting duplicated data, and then stops the replication task.
-    on-duplicate: "replace"
+
+    # The import mode during the full import phase. The following modes are supported:
+    # - "logical" (default). Uses TiDB Lightning's logical import mode to import data. Document: https://docs.pingcap.com/tidb/stable/tidb-lightning-logical-import-mode
+    # - "physical". Uses TiDB Lightning's physical import mode to import data. Document: https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-physical-import-mode
```
Comment (ran-huang) [inline dm/task-configuration-file-full.md:126]:
```suggestion
    # - "physical". Uses TiDB Lightning's physical import mode to import data. Document: https://docs.pingcap.com/zh/tidb/stable/tidb-lightning-physical-import-mode
    #   The "physical" mode is still an experimental feature and is not recommended in production.
```

**Review by lance6716 — APPROVED** [review]

**Pull request merged**